cql3/restrictions/statement_restrictions.cc

```diff
@@ -992,7 +992,9 @@ namespace {
 using namespace expr;
 
 /// Computes partition-key ranges from token atoms in ex.
-dht::partition_range_vector partition_ranges_from_token(const expr::expression& ex,
```

Comment:
> The catesean product of a large query range with

Don't we have a limit on cartesian product sizes?

Reply:
Good point.
`get_partition_key_ranges` seems safe indeed.

`indexed_table_select_statement::do_execute_base_query` concurrency limit is 4096.
With which, `ranges_to_vnodes` may be asked to return a vector of 4096 partition ranges.
With 120 bytes each, that gets us to 491520 bytes which is a large allocation, but not terribly large.

Cc @raphaelsc